MoreLinq/Experimental/SelectAsync.cs

```diff
@@ -0,0 +1,548 @@
+                            // is the next that should be yielded then yield
+                            // the result.
+
+                            yield return value;
+
+                            if (holds != null) // preserve order?
+                            {
+                                // Release withheld results consecutive in key
+                                // order to the one just yielded...
+
+                                var releaseCount = 0;
+
+                                for (nextKey++; holds.Count > 0; nextKey++)
+                                {
+                                    var (candidateKey, candidate) = holds[0];
+                                    if (candidateKey == nextKey)
+                                    {
+                                        releaseCount++;
+                                        yield return candidate;
+                                    }
+                                }
+
+                                holds.RemoveRange(0, releaseCount);
+                            }
+                        }
+                        else
+                        {
+                            // Received a result out of order when order must be
+                            // preserved, so withhold the result by finding out
+                            // where it belongs in the order of results withheld
+                            // so far and insert it in the list.
+
+                            var i = holds.BinarySearch(result, TupleComparer<int, TResult>.Key1);
+                            Debug.Assert(i < 0);
+                            holds.Insert(~i, result);
+                        }
+                    }
+
+                    if (holds?.Count > 0) // yield any withheld, which should be in order...
+                    {
+                        foreach (var (key, value) in holds)
+                        {
+                            Debug.Assert(nextKey++ == key); //...assert so!
+                            yield return value;
+                        }
+                    }
+
+                    completed = true;
+                }
+                finally
+                {
+                    // The cancellation token is signaled here for the case where
+                    // tasks may be in flight but the user stopped the enumeration
+                    // partway (e.g. SelectAsync was combined with a Take or
+                    // TakeWhile). The in-flight tasks need to be aborted as well
+                    // as the awaiter loop.
+
+                    if (!completed)
+                        cancellationTokenSource.Cancel();
+                    disposable.Dispose();
+                }
+            }
+        }
+
+        enum Notice { Result, Error, End }
+
+        static async Task<TResult> Select<T, TResult>(this Task<T> task, Func<T, TResult> selector) =>
+            selector(await task);
```

Comment:
I'm not an expert in async code, but I think it is best practice to always use `ConfigureAwait(false)` in library code.